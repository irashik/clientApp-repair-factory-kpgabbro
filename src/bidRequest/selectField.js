import React, { useState } from 'react';
import Form from 'react-bootstrap/Form';

function StatusSelectField(props) {

    function handleStatusSelect(e) {
        props.handleStatus(e);
    }
    
    return (
        <Form.Control as='select' size="sm" aria-label="Выберите статус заявки"
            value={props.statusBid} id='filterSelectStatusBid'
            onChange={(e) => handleStatusSelect(e.target.value)}  >
                
            <option value=''>Выберите статус</option>
            <option value="DRAFT">Черновик</option>
            <option value="CANCELLED">Отменено</option>
            <option value="FINISHED">Завершено</option>
            <option value="DEFERRED">Отложено</option>
            <option value="INWORK">В работе</option>
            <option value="NEW">Новая</option>
            {/* Заказано, отправлено, согласование ... */}

        </Form.Control>  
    );

};
export {StatusSelectField}; 


function CategorySelectField(props) {

    function handleCategorySelect(e) {
        props.handleCategory(e);
    }

    return (
        <Form.Control   as="select" 
                        size="sm" 
                        aria-label="Выберите категорию"
                        value={props.category}
                        onChange = { (e) => handleCategorySelect(e.target.value)}
                        placeholder="Выберите категорию"
                        id='filterSelectCategoryBid'>

                        <option value='' >Выберите категорию</option>
                        <option value="Инструмент">Инструмент</option>
                        <option value="Расходники">Расходники</option>
                        <option value="Запчасти">Запчасти</option>
                        <option value="Прочее">Прочее</option>
                    </Form.Control>
    )
};
export {CategorySelectField};

function PrioritySelectField(props) {
  
    function handlePrioritySelect(e) {
        props.handlePriority(e);
    }

    return (
            <Form.Control   as="select" size="sm" 
                            aria-label="Выберите приоритет"
                            id='filterSelectPriorityBid'
                            onChange={(e) => handlePrioritySelect(e.target.value)}
                            value={props.priority} >

                <option value=''>Выберите приоритет</option>
                <option value="Срочно">Срочно!</option>
                <option value="Планово">Планово</option>
                <option value="Желательно">Желательно</option>
            </Form.Control>
    )

};
export {PrioritySelectField};
