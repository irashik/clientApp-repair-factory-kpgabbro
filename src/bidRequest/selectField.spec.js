


import React from 'react';
import { shallow } from 'enzyme';

import {StatusSelectField, CategorySelectField, PrioritySelectField } from './selectField';



describe('MatchSnapShot components', () => {

    test('StatusSelectField snapShot', () => {

        const component = shallow(<StatusSelectField />);
        expect(component).toMatchSnapshot();

    });

    test('CategorySelectField snapShot', () => {

        const component = shallow(<CategorySelectField />);
        expect(component).toMatchSnapshot();

    });

    test('PrioritySelectField snapShot', () => {

        const component = shallow(<PrioritySelectField />);
        expect(component).toMatchSnapshot();

    });


});


