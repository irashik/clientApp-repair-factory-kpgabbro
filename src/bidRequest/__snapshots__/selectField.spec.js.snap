// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchSnapShot components CategorySelectField snapShot 1`] = `
<FormControl
  aria-label="Выберите категорию"
  as="select"
  id="filterSelectCategoryBid"
  onChange={[Function]}
  placeholder="Выберите категорию"
  size="sm"
>
  <option
    value=""
  >
    Выберите категорию
  </option>
  <option
    value="Инструмент"
  >
    Инструмент
  </option>
  <option
    value="Расходники"
  >
    Расходники
  </option>
  <option
    value="Запчасти"
  >
    Запчасти
  </option>
  <option
    value="Прочее"
  >
    Прочее
  </option>
</FormControl>
`;

exports[`MatchSnapShot components PrioritySelectField snapShot 1`] = `
<FormControl
  aria-label="Выберите приоритет"
  as="select"
  id="filterSelectPriorityBid"
  onChange={[Function]}
  size="sm"
>
  <option
    value=""
  >
    Выберите приоритет
  </option>
  <option
    value="Срочно"
  >
    Срочно!
  </option>
  <option
    value="Планово"
  >
    Планово
  </option>
  <option
    value="Желательно"
  >
    Желательно
  </option>
</FormControl>
`;

exports[`MatchSnapShot components StatusSelectField snapShot 1`] = `
<FormControl
  aria-label="Выберите статус заявки"
  as="select"
  id="filterSelectStatusBid"
  onChange={[Function]}
  size="sm"
>
  <option
    value=""
  >
    Выберите статус
  </option>
  <option
    value="DRAFT"
  >
    Черновик
  </option>
  <option
    value="CANCELLED"
  >
    Отменено
  </option>
  <option
    value="FINISHED"
  >
    Завершено
  </option>
  <option
    value="DEFERRED"
  >
    Отложено
  </option>
  <option
    value="INWORK"
  >
    В работе
  </option>
  <option
    value="NEW"
  >
    Новая
  </option>
</FormControl>
`;
